### PR TITLE
Update acme-lb with example-gatewayClass.

### LINKED ISSUE #2697: The User Guide lacks information for beginners
The Guides require knowledge about the gateway api resources without explaining it.
For example the [Simple Gateway](https://gateway-api.sigs.k8s.io/guides/simple-gateway/) guide assumes a gateway class `acme-lb`. There is no explaination what a gateway class is, what it does, why i need it and how it is created. 

As a beginner I want a getting started guide which I can work through from the beginning and have a proper understanding at the end. It falls a bit into the second todo in #558 about having the guides create a real deployment instead of just theoretical steps. Having a guide actually creating something on a testsystem without assuming anything (apart maybe from basic kubernetes stuff, but even that could be done shortly in the first step), would benefit beginners and speed up their understanding of how the parts work with each other

### PR DESCRIPTION
**Pull request by sawsa307**
* Update acme-lb with example-gatewayClass to make it more obvious that it is a placeholder gatewayClass that is intended to be replaced.
* Add explanations and reference in simple-gateway.md on Gateway and GatewayClass concepts.

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Make it more clear that `acme-lb` is meant to be a placeholder, and add reference to a list of available GatewayClass.

**Which issue(s) this PR fixes**:
Fixes #2697

**Does this PR introduce a user-facing change?**:
No, it is a doc change.

### CODE REVIEW

File: examples/experimental/basic-grpc.yaml [hunk 1]
```diff
@@ -3,11 +3,11 @@
 apiVersion: gateway.networking.k8s.io/v1
 kind: GatewayClass
 metadata:
-  name: acme-lb
+  name: example-gatewayClass
```
Comment (robscott) [inline examples/experimental/basic-grpc.yaml:6]: Maybe just use "example"

File: site-src/api-types/grpcroute.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ Each Route includes a way to reference the parent resources it wants to attach
 to. In most cases, that's going to be Gateways, but there is some flexibility
 here for implementations to support other types of parent resources.
 
-The following example shows how a Route would attach to the `acme-lb` Gateway:
+The following example shows how a Route would attach to the `example-gatewayClass` Gateway:
```
Comment (robscott) [inline site-src/api-types/grpcroute.md:81]: You're only changing the name of the GatewayClass in examples, not Gateways.
```suggestion
The following example shows how a Route would attach to the `acme-lb` Gateway:
```

File: site-src/api-types/grpcroute.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ metadata:
   name: grpcroute-example
 spec:
   parentRefs:
-  - name: acme-lb
+  - name: example-gatewayClass
```
Comment (robscott) [inline site-src/api-types/grpcroute.md:90]: Same comment as above, would revert this change.

File: site-src/api-types/httproute.md [hunk 4]
```diff
@@ -29,15 +29,15 @@ Each Route includes a way to reference the parent resources it wants to attach
...
+The following example shows how a Route would attach to the `example-gatewayClass` Gateway:
 ```yaml
 apiVersion: gateway.networking.k8s.io/v1beta1
 kind: HTTPRoute
 metadata:
   name: httproute-example
 spec:
   parentRefs:
-  - name: acme-lb
+  - name: example-gatewayClass
```
Comment (robscott) [inline site-src/api-types/httproute.md:40]: Same as above

File: site-src/guides/simple-gateway.md [hunk 5]
```diff
@@ -12,11 +12,18 @@ match all HTTP traffic and directs it to a single Service named `foo-svc`.
...
 ```
 
-The Gateway represents the instantiation of a logical load balancer. It's
-templated from a hypothetical `acme-lb` GatewayClass. The Gateway listens for
-HTTP traffic on port 80. This particular GatewayClass automatically assigns an
-IP address which will be shown in the `Gateway.status` after it has been
-deployed. 
+The Gateway represents the instantiation of a logical load balancer and the
+GatewayClass defines the load balancer template when users create a Gateway.
+The example Gateway is templated from a hypothetical `example-gatewayClass`
```
Comment (robscott) [inline site-src/guides/simple-gateway.md:17]:
```suggestion
The example Gateway is templated from a hypothetical `example`
```

File: site-src/guides/simple-gateway.md [hunk 6]
```diff
@@ -12,11 +12,18 @@ match all HTTP traffic and directs it to a single Service named `foo-svc`.
...
+GatewayClass, which is meant to be a placeholder and substitued by users. Here
+is a list of available
+[Gateway Implementation](https://gateway-api.sigs.k8s.io/implementations/) that
+can be use to determine the correct GatewayClass based on the specific
```
Comment (robscott) [inline site-src/guides/simple-gateway.md:21]:
```suggestion
can be used to determine the correct GatewayClass based on the specific
```

**Review by robscott — COMMENTED** [review]
Thanks @sawsa307!

**Review by shaneutt — APPROVED** [review]
/ok-to-test
/approve
/cc @robscott
